integration/test_installation.py

```diff
@@ -0,0 +1,79 @@
+import time
+import os
+import unittest
+
+from mozlog import get_default_logger
+
+from marionette import Marionette
+from marionette_driver import Actions, Wait
+from marionette_driver.by import By
+from marionette_driver.addons import Addons
+from marionette_driver.errors import NoSuchElementException
+
+from firefox_puppeteer.api.prefs import Preferences
+from firefox_puppeteer.api.software_update import SoftwareUpdate
+from firefox_puppeteer.ui.update_wizard import UpdateWizardDialog
+from firefox_puppeteer.ui.browser.notifications import (
+    AddOnInstallBlockedNotification,
+    AddOnInstallFailedNotification,
+    AddOnInstallConfirmationNotification,
+    AddOnInstallCompleteNotification,
+)
+
+from __init__ import FirefoxTestCase
+
+
+SITE_URL = 'http://testpilot.dev:8000/'
+TOOLBAR_BUTTON_ID = 'toggle-button--testpilot-addon-testpilot-link'
+
+
+class TestAddonInstallation(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+    def test_initial_installation(self):
+        m = self.marionette
+        b = self.browser
+
+        # Navigate to the site and click the install button
+        with m.using_context(m.CONTEXT_CONTENT):
+            m.navigate(SITE_URL)
+            m.find_element(By.CSS_SELECTOR,
```

Comment:
This line is the reason I mentioned the search timeout above: it's possible for find_element to give up before the front-end JS has a chance to template the page.